ui/components/app/modals/confirm-delete-network/confirm-delete-network.component.js

```diff
@@ -14,11 +14,10 @@ export default class ConfirmDeleteNetwork extends PureComponent {
     t: PropTypes.func,
   };
 
-  handleDelete = () => {
-    this.props.delRpcTarget(this.props.target).then(() => {
-      this.props.onConfirm();
-      this.props.hideModal();
-    });
+  handleDelete = async () => {
+    await this.props.delRpcTarget(this.props.target);
+    this.props.onConfirm();
+    this.props.hideModal();
```

Comment:
just to note that this changes the return type of this function from 'void' to 'Promise<void>' Its been a hot minute since I dived into this but is it fine to return promises from things like html property click handlers etc? I did a quick search and its definitely prevalent, but i'm curious how TypeScript and React.MouseEvent types handle promise returning handlers.

I'm not blocking on this cause its more of a curiosity of mine rather than a request for any sort of change.